app/models/compute_resource.rb

```diff
@@ -77,6 +77,10 @@ def self.providers_requiring_url
     _("Libvirt, oVirt, OpenStack and Rackspace")
   end
 
+  def self.providers_url_mandatory
+    %w(libvirt ovirt openstack rackspace)
```

Comment:
Frankly I'd squash this with `is_eligible_for_url?` method, unless you have specific reason to have it like that.